cmd/authelia-scripts/cmd_docker.go

```diff
@@ -83,21 +93,60 @@ func deploy(docker *Docker, tag string) {
 	}
 }
 
+func deployManifest(docker *Docker, tag string, amd64tag string, arm32v7tag string, arm64v8tag string) {
+	imageWithTag := DockerImageName + ":" + tag
+	fmt.Println("===================================================")
+	fmt.Println("Docker manifest " + imageWithTag + " will be deployed on Dockerhub.")
+	fmt.Println("===================================================")
+
+	err := docker.Tag(DockerImageName, imageWithTag)
+
+	if err != nil {
+		panic(err)
+	}
+
+	docker.Manifest(imageWithTag, amd64tag, arm32v7tag, arm64v8tag)
+
+	if err != nil {
+		panic(err)
+	}
+}
+
 func publishDockerImage() {
 	docker := &Docker{}
 
+	ARCH := os.Getenv("ARCH")
```

Comment:
I'd prefer having a flag for this option so that it is documented in the usage of the command. The other environment variables are env vars because they explicitly come from Travis and do not need to be known by the user to run the command. This is not the case or this arch customization, and the users should be aware of it from the usage to avoid running the command without the flag and get an erroneous name of their docker image.

You can use amd64 as the default value if the flag is not provided.

Reply:
As discussed in previous thread, this script is for users and so must have a human friendly interface even if it can (and is) be called from the CI. An optional flag is a human friendly way of configuring the script and documenting the feature to users. Moreover it does not cost anything compared to the configuration by env variable. Regarding your concern about not setting the arch, I'd say that if a user omit it let's use amd64 as the default value.

Does it sound reasonable?

Reply:
The `ARCH` is only utilised when looking to publish images to Docker, which as far as I can see is only within your Travis build pipeline, because it would not execute if the respective Travis variables are not present.

The `ARCH` variable isn't really what's going to cause issues for you as a user, it's the fact that we're now explicitly passing the `Dockerfile` into the helper function for builds.
I've now specified a default value in this helper file which will revert to using `Dockerfile` being the amd64 variant if the `DOCKERFILE` variable isn't present (outside of Travis builds).

This should allow you to continue as per usual and support the broader multistage builds for Travis too.